web_server/src/workers/job.rs

```diff
@@ -0,0 +1,168 @@
+            .await
+            .with_context(|| "Could not connect to SQLite in-memory")
+            .unwrap();
+        SqliteStorage::setup(&pool)
+            .await
+            .expect("Could not run sqlite migrations");
+        let storage: SqliteStorage<SanitisePDFRequest> = SqliteStorage::new(pool);
+        let mutex_storage = Mutex::new(storage.clone());
+
+        let file_storage = FileStorage::new(settings.sanitisation.pdfs_dir);
+        let file_storage = Arc::new(file_storage);
+
+        let monitor = Monitor::new().register({
+            WorkerBuilder::new("pdf-regenerator")
+                .enable_tracing()
+                .data(file_storage)
+                .concurrency(10)
+                .backend(storage)
+                .build_fn(sanitise_pdf)
+        });
+
+        Ok(Self {
+            storage: mutex_storage,
+            monitor: Mutex::new(Some(monitor)),
+        })
+    }
+
+    #[instrument(skip(self))]
+    pub async fn enqueue(&self, job: SanitisePDFRequest) -> Result<()> {
+        let mut guard = self.storage.lock().await;
+        guard
+            .push(job.clone())
+            .await
+            .map_err(|e| anyhow!("Could not enqueue job for processing. job={job:#?} error={e}"))?;
+
+        Ok(())
+    }
+
+    pub async fn run_until_stopped(&self) -> std::io::Result<()> {
+        let mut guard = self.monitor.lock().await;
+        match guard.take() {
+            Some(monitor) => monitor.run_with_signal(signal::ctrl_c()).await,
+            None => Err(std::io::Error::other(
+                "Can only run the job scheduler once in the lifecycle of the server",
+            )),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ProcData {
+    original_file: String,
+    output_file: String,
+}
+
+#[instrument(skip(data))]
+async fn sanitise_pdf(
+    job: SanitisePDFRequest,
+    data: apalis::prelude::Data<Arc<FileStorage<String>>>,
+) -> Result<(), JobError> {
+    let fut = tokio::task::spawn_blocking(move || {
+        tracing::info!("Processing PDF. filename={}", job.filename);
+        let original_file = Path::new(data.base_dir()).join(&job.filename);
+        let output_file = Path::new(data.base_dir()).join(format!("processed_{}", &job.filename));
+
+        let args = ProcData {
+            original_file: original_file.to_str().unwrap().into(),
+            output_file: output_file.to_str().unwrap().into(),
+        };
+
+        let proc_handle = procspawn::spawn(args, |args| {
+            match regenerate_pdf(&args.original_file, &args.output_file) {
+                Ok(()) => {
+                    tracing::info!("File regenerated successfuly");
+                    None
+                }
+                Err(error) => Some(format!(
+                    "Failed to regenerate file. filename={} error={}",
+                    args.original_file, error
+                )),
+            }
+        });
```

Comment:
procspawn supports a pool for processes so we don't have to be spawning them constantly.